blockchain/chain_sync/src/tipset_syncer.rs

```diff
@@ -1163,16 +1162,16 @@ async fn validate_tipset<DB: Blockstore + Store + Clone + Send + Sync + 'static,
     debug!("Tipset keys: {:?}", full_tipset_key.cids);
 
     for b in blocks {
-        let validation_fn = tokio::task::spawn(validate_block::<_, C>(
-            consensus.clone(),
-            state_manager.clone(),
-            Arc::new(b),
-        ));
-        validations.push(validation_fn);
+        let consensus = consensus.clone();
+        let state_manager = state_manager.clone();
+        let validation_fn = tokio::task::spawn_blocking(|| {
```

Comment:
Any reason to move on a `spawn_blocking` here? `validate_block` is an async function, no?

Reply:
tokio-console shows warning when using spawn with a "⚠ n tasks have lost their waker" at this line. Replacing spawn with spawn_blocking makes the warns go away.

**Update**: I no longer receive those warns even with `spawn` after repeated runs with tokio-console, so I'm thinking of reverting this change.

Reply:
Not sure about the real root cause of this "have lost their wake" warning. Some other people are complaining of it too, see https://github.com/tokio-rs/console/issues/149.